File: Bai_101/main.py
import math

def is_prime(num):
    if num <= 1: return False
    if num <= 3: return True
    if num % 2 == 0 or num % 3 == 0: return False
    i = 5
    while i * i <= num:
        if num % i == 0 or num % (i + 2) == 0:
            return False
        i += 6
    return True

def check(n):
    mp = {}
    e = p = 0
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            e += 1
            while n % i == 0:
                if i in mp: mp[i] += 1
                else: mp[i] = 1
                n //= i
                p += 1
    
    if n != 1:
        e += 1
        p += 1
        mp[n] = 1
    
    if e >= 3:
        print("YES")
        res = 1
        i = 0
        for key in mp:
            i += 1
            if i <= 2:
                print(key, end=' ')
                mp[key] -= 1
            res *= key ** mp[key]
        print(res)
        return
    
    if e == 2 and p >= 4:
        print("YES")
        res = 1
        for key in mp:
            print(key, end=' ')
            res *= key ** (mp[key] - 1)
        print(res)
        return
    
    if e == 1 and p >= 6:
        print("YES")
        key = list(mp.keys())[0]
        print(key, key ** 2, key ** (mp[key] - 3))
        return
    
    print("NO")

def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        if is_prime(n):
            print("NO")
        else:
            check(n)

if __name__ == "__main__":
    main()

File: Bai_101/main.cpp
#include<bits/stdc++.h>
using namespace std;

using ll = long long;

bool isPrime(int num){
    if(num <= 1) return false;
    if(num <= 3) return true;
    if(num % 2 == 0 || num % 3 == 0) return false;
    for(int i = 5; i * i <= num; i += 6)
        if(num % i == 0 || num % (i + 2) == 0)
            return false;
    return true;
}

void check(int n){
    // e : số lượng ước
    // p : tổng số mũ của các ước
    map<int,int> mp;
    int e = 0, p = 0;
    for(int i = 2; i <= sqrt(n); ++i){
        if(n % i == 0){
            ++e;
            while(n % i == 0){
                mp[i]++;
                n /= i;
                p++;
            }
        }
    }
    
    if(n != 1){
        e++;
        p++;
        mp[n]++;
    }

    if(e >= 3){
        cout << "YES\n";
        ll res = 1;
        int i = 0;
        for(auto it : mp){
            ++i;
            if(i <= 2){
                cout << it.first << ' ';
                it.second -= 1;
            }
            res *= pow(it.first, it.second);
        }
        cout << res << endl;
        return;
    }
    if(e == 2 and p >= 4){
        cout << "YES\n";
        ll res = 1;
        for(auto it : mp){
            cout << it.first << ' ';
            res *= 1ll * pow(it.first, it.second - 1);
        }
        cout << res << endl;
        return;
    }
    if(e == 1 and p >= 6){
        cout << "YES\n";
        auto it = mp.begin();
        cout << it->first << ' ' << 1ll * pow(it->first,2) << ' ' << 1ll * pow(it->first, it->second - 3) << endl;
        return;
    }
    cout << "NO";
}

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int t; cin >> t;
    while(t--){
        int n; cin >> n;
        if(isPrime(n)) cout << "NO\n"; 
        else check(n);
    }
    return 0;
}